app/src/auto_validator/core/models.py

```diff
@@ -22,64 +28,53 @@ def url(self):
 
 
 class Block(models.Model):
-    serial_number = models.IntegerField(primary_key=True, unique=True)  # Unique identifier for the block
-    timestamp = models.DateTimeField()  # Time when the block was created
+    serial_number = models.IntegerField(primary_key=True, unique=True)
+    timestamp = models.DateTimeField()
 
     def __str__(self):
-        return f"Block {self.serial_number} at {self.timestamp}"
+        return f"{self.serial_number}"
 
 
 class Subnet(models.Model):
     name = models.CharField(max_length=255)
-    description = models.TextField(null=True, blank=True)  # Additional field for description
-    operators = models.ManyToManyField("Operator", related_name="subnets", blank=True)  # Link to operators
-    running_mainnet = models.BooleanField(default=False)
-    running_testnet = models.BooleanField(default=False)
+    description = models.TextField(null=True, blank=True)
+    operators = models.ManyToManyField("Operator", related_name="subnets", blank=True)
 
-    def short_registration_indicator(self):
-        mainnet_slots = self.slots.filter(blockchain="mainnet")
-        testnet_slots = self.slots.filter(blockchain="testnet")
+    def registered_networks(self):
+        mainnet_slots = self.slots.filter(blockchain="mainnet", registration_block__isnull=False)
+        testnet_slots = self.slots.filter(blockchain="testnet", deregistration_block__isnull=False)
 
         mainnet_indicator = f"sn{mainnet_slots.first().netuid}" if mainnet_slots.exists() else ""
         testnet_indicator = f"t{testnet_slots.first().netuid}" if testnet_slots.exists() else ""
 
-        if mainnet_indicator and testnet_indicator:
-            return f"{mainnet_indicator}{testnet_indicator}"
-        elif mainnet_indicator:
-            return mainnet_indicator
-        elif testnet_indicator:
-            return testnet_indicator
-        else:
-            return "-"
+        return f"{mainnet_indicator} {testnet_indicator}" or "-"
 
     def __str__(self):
         return self.name
 
 
 class SubnetSlot(models.Model):
-    subnet = models.ForeignKey(Subnet, on_delete=models.CASCADE, related_name="slots")
+    subnet = models.ForeignKey(Subnet, on_delete=models.SET_NULL, null=True, blank=True, related_name="slots")
```

Comment:
I think it's protect again - at least in the initial version we'd like the user to go delete the server, then subnetslot, then subnet, not everything automatically. One day we'll automate it fully so that when a subnet deregs, the server is decomissioned automatically, subnet slot is marked as deregged etc - in the first version lets stay on the side of caution